### DISCUSSION TITLE
Conductor loss for wideband models

### DISCUSSION THREAD

**Discussion by biergaizi** [thliebig/openEMS-Project#161] [Ideas]
Recently, several members from this forum have studied the problem of modeling dielectric materials with frequency-dependent permittivity in openEMS. Thanks to the development by @toammann, a promising solution has already been demonstrated.

However, we still don't have a satisfactory solution to a closely-related conductor loss problem. In particular, to match experimental results, we need two possible compensations in the model, including the effects of *surface roughness* and *surface finish*. 

The idea and context of this thread originated from [Some Simulation Results with Wideband Dielectric models](https://github.com/thliebig/openEMS-Project/discussions/157). To improve clarify of the discussion, this thread is the new catchall place to discuss problems specifically related to lossy conductor modeling.

**Comment by biergaizi** [comment]
The followings are my research notes on surface roughness modeling. Everyone is free to introduce any other related topic by creating a separate comment (for ease of navigation, reply to this comment only when it's relevant to its content).

---

## Notes on Surface Roughness Modeling and Measurement

**The Need of Surface Roughness Modeling**: Accurate prediction of attenuation for microwave and high-speed digital circuit board design requires modeling surface roughness of the copper foil.

### Parameters

1. **Incompatible Parameters**: Unfortunately, many different parameters are used from incompatible measurement methods, worse, different roughness models requires different parameters as their inputs.

2. **Profile Parameters**: A roughness value can either be calculated on a profile (line) or on a surface (area). The profile roughness parameters are more common, such as Ra, Rq, and Rz and are usually measured by by a mechanical profilometer.

    2.1. **Ra, Rq, and Rz**: Ra is the absolute average of the profile values, Rq is the root mean square (RMS) of the profile values, and Rz is the maximum peak to valley height.

    2.2. **Ra** is common in material datasheet but it's not an ideal figure of merit, its uses are historical as it predated other modern measurement methods [6]. Rq and Rz are needed for circuit board modeling.

    2.3. **Three Rz measurement methods**: Rz is measured by three different standards: DIN, JIS and ISO. DIN averages "five highest peaks and five lowest valleys over a sample length." JIS averages "five highest peak and five lowest valleys over *five* sample length". ISO uses "the maximum peak to valley distance over a sample length.". DIS and JIS's standards are compatible in general, while Polar Instruments doesn't recommend ISO's method due to small number of data points [3].

3. **Surface Parameters**: The area roughness parameters are denoted as Sa, Sq, etc. These values can be measured by white-light interferometry. They're less less common, but is the prefered method by Rogers Corporation who claims they have better predictive power [5].

4. **Huray Snowball Parameters**: Huray's "snowball" models surface roughness as metal snowballs on the conductor surface. The parameters are *Ratio of Areas*, *Effective Ball Radius (um)*, *Area of Ball Count (sq um)*, and *Number of Balls in Area*. It's theoretically possible to derive these values from first principles by analyzing the images using an electron microscope. But practically speaking, these values are fitted from either electron microscope images or measured attenuation.

## Models

1. **Hammerstad**: Hammerstad and Jensen developed a series of standard formulas widely used in textbooks and calculators, including surface roughness correction. It models surface roughness as a sawtooth pattern on the surface of the conductor, and only takes a single input parameter, *Rq* (RMS profile values). Despite its popularity, it's "only valid up to a few GHz" as Polar Instruments claims. Furthermore, researchers have found that the model is only accurate when surface roughness is relatively low, it "saturates" under high roughness and underestimates attenuation. [1]

2. **Groisse**: An alternative model that suffers from limitations similar to Hammerstad's model, the input parameters is also *Rq*.

3. **Huray**: To overcome the limitation of existing model, Huray proposed a physics-first model based on electron microscope images. Rather than modeling surface roughness is a sawtooth pattern, it models roughness as metal "snowballs" on the conductor surface. This model's accuracy has been widely accepted by the industry - as long as you can figure out how to fit the complicated parameters involved.

4. **Cannonball-Huray Model**: The parameters of the Huray's model must be determined empirically with specialized test fixture and careful fitting. Thus, it's often inaccessible to the average circuit board designer. Bert Simonovich proposed a method [2] to find an approximated version of Huray parameters, using just Rz (maximum peak to valley height) as input by assuming the "snowballs" are organized according to close-packing of equal spheres (also known as the Cannonball Theorem in geometry).

    4.1. **Validity and Usage**: There have been some controversy about the theoretical foundation of this model [4], but the author insists that it's an useful rule of thumb when only datasheet Rz values are available even if it's only a rough approximation (no pun intended). This model is included in Si9000, an widely-used transmission line calculator and field solver in the PCB manufacturing industry.

## Bibliography

1. [*Surface Roughness Effect*](https://www.oldfriend.url.tw/HFSS/ansys_en_Roughness.html)

    Great overview of this topic, reading is strongly recommended. 

2. [*Cannonball-Huray Model Demystified*](https://blog.lamsimenterprises.com/2019/03/29/cannonball-huray-model-demystified/)

    From the author of the Cannonball-Huray model, reading is strongly recommended.

3. [*Choosing the right Surface Roughness Measurement*](https://www.polarinstruments.com/support/si/AP8206.html), Polar Instruments

4. [*Conductor surface roughness modeling: From “snowballs” to “cannonballs”*](https://www.simberian.com/AppNotes/SnowballsToCannonballs.pdf), Yuriy Shlepnev, Simberian Inc.

5. [*Copper Foils for High Frequency Materials*](https://www.rogerscorp.cn/-/media/project/rogerscorp/documents/advanced-electronics-solutions/english/properties---detailed-characteristics/copper-foils-for-high-frequency-circuit-materials.pdf), Rogers Corporation

6. https://en.wikipedia.org/wiki/Surface_roughness#Profile_roughness_parameters

**Reply by toammann** [reply]
Hello,
I want to add another model to the discussion, the Gradient Model. It is a fairly new one and only a view years old. I came across it because it is implemented in newer versions of CST Microwave studio by default.

The author starts right at the beginning of mexwell's equations. So, it has a solid physical justification. The surface roughness is is characterized statistically by a CDF (Cumulative Distribution Function). He showed measured results that justify that the roughness can be represented by a gaussian distribution. Therefore, only  one parameter is required to characterize the roughness - the Rq (Root-Mean-Square) value. For gaussian distributions this is also the standard deviation.

This is in my opinion the greatest advantage of this model. It is very hard to get reliable values for surface roughness from board manufacturers. Models like the Huray model require multiple input values than cannot be measured directly. The Rq value on the other hand can be extracted from a microsection cut of a transmission line. With a reasonable microscope this can even be done at home (It would be a cool experiment to test this!).

The second huge advantage of this model is that phase effects are handled properly. The boundary between the dielectric and a rough surface is not aprupt. In the transition area exist microscopic effects that cause a macroscopic slowing of the wave. This is the reason for the often encountered "Design-DK". 

Unfortunately, I have never done a proper verification of the model accuracy. The measurements published by the author look amazing! It would be a very cool project to verify this (but this is a future topic).

If someone wants to read into the model (I am also still in the process). I can recommend the following papers.

A paper about conductor losses, "Design-DK" and Epsilon apperent from 2012:
[[1] Measuring Design-DK and true permittivity of PCB materials up to 20GHz](https://ieeexplore.ieee.org/document/7107776)

A summary on the Gradient model from 2017 
[[2] A Physical Surface Roughness Model and Its Applications](https://ieeexplore.ieee.org/document/7932148)

To summarize the Gradient Model features:
- Justified by mexwells equations
- Only one easily measurable parameter for roughness (Rq, standard deviation)
- Takes the phase effects into acount
- Already implemented in commercial solvers by default

Regards!

**Comment by biergaizi** [comment]
Another source of error is the losses from gold plating. Losses from the ENIG surface finish has a magnitude comparable to surface roughness due to the use of nickel and gold. See:

* D. Cullen, B. Kline, G. Moderhock, and L. Gatewood, "Effects Of Surface Finish On High Frequency Signal Loss Using Various Substrate Materials", July 2001. https://smtnet.com/library/index.cfm?fuseaction=view_article&article_id=3376&company_id=59005

* X. Wu, D. Cullen, G. Brist and O. M. Ramahi, "Surface Finish Effects on High-Speed Signal Degradation," in IEEE Transactions on Advanced Packaging, vol. 31, no. 1, pp. 182-189, Feb. 2008, doi: 10.1109/TADVP.2007.914962.  https://ieeexplore.ieee.org/document/4437763

Although I'm afraid that it may only change the result by 0.1 dB or so even when considering ENIG, but for further verification, It's worthwhile to run a simulation with both surface roughness and ENIG losses turned on. Some commercial simulators have lossy surface finish models.  I'm pretty sure that closed-form approximation formulas also exist.

**Comment by toammann** [comment]
Hello all,

I have started to think about the conductor loss problem. The more I think and read about the more complicated it gets. 

First, I was under the impression that it is sufficient to fit a Drude model to one of the roughness models above. In the end the roughness model must be made available to the simulator in some way e.g. in the form of an equivalent conductivity which can (maybe) be approximated by a Drude model. However, I have now realized that this will still be comparable to the "normal" type of material where the skin effect must be meshed. 

I think the easiest way to explain my thoughts is by neglecting the roughness discussion for now. In fact the gradient model approaches a smooth surface for Rq=0. The results are then identical to the classical skin effect we all know from university. Even the imaginary part of the Drude model converges for low frequencies ($\omega << \omega_P$) towards the "normal" material model which [is in openems](https://wiki.openems.de/index.php/Material_Property.html) where $\kappa$ is assumed to be DC conductivity:

![image](https://github.com/thliebig/openEMS-Project/assets/9382233/7732f07a-5a94-4dcc-9787-807311f86298)

Therefore, the normal material model covers the classical skin effect for a smooth metal already. Of course the real part should not be equal to $\epsilon_R = 1$ but the value is at this point not important for the discussion. 

I think the following conclusions can be made:

1.  The normal material model covers the skin effect already. However a very fine mesh resolution is required
2. Any fit we perform on the Drude model will suffer from the limitation of 1). The material is handled correctly by the simulator but the mesh must be fine enough to resolve the pyscial behavior of the metal.
3. We should not focus on copper roughness at this point. Any fit of a roughness model to a Drude model will require fine meshing
4. What we actually want is a description of the metal that covers skin effect and copper roughness without the need to mesh them

A concept that enables option 4) is the surface impedance concept. The tangential E- and H-fields on the metallic top surface can be related by a surface impedance Z:

![image](https://github.com/thliebig/openEMS-Project/assets/9382233/33c54a85-59e5-45cd-9cda-e297bcf451d9)

The tangential fields (the metal assumed in the yz-plane), cause an "equivalent" current on the top surface. The surface impedance is chosen in a what such that this equivalent current is identical to the real current that would flow in reality with skin effect. However, only the tangential fields at the metal boundary must be evaluated. The metals internal don´t need to be meshed.

I think that implementing a concept like this would require a new material in openEMS along to PEC normal and Lorentz. No idea what is required for implementation at this point. 

However, this concept can be quite powerful. We can than (more or less) easily add surface roughness to the simulation simply by providing different values for the surface impedance.

So the very coarse steps I am proposing are:

1. Check if the surface impedance concept is feasible for openEMS or if we want to go down this road
2. Implement the surface impedance for metals (Time domain fit by equivalent circuit?)
3. Work through the Gradient model papers, solve the differential equation numerically in octave

I am not experienced in this topic but I found it very interesting to read about these topics. Please feel free to correct me if some of the above is not correct!

Regards,
Tobias

**Reply by biergaizi** [reply]
> What we actually want is a description of the metal that covers skin effect and copper roughness without the need to mesh them

Right, it's also what I was thinking. It would be the best if this can be implemented as a small modification of the existing Conducting Sheet model (which is based on the paper [1]), which already models surface impedance but without considering roughness's contributions. But I have no idea about whether it's feasible.

[1] Lauer, A., & Wolff, I. (n.d.). *A conducting sheet model for efficient wide band FDTD analysis of planar waveguides and circuits.* 1999 IEEE MTT-S International Microwave Symposium Digest (Cat. No.99CH36282). doi:10.1109/mwsym.1999.780262

**Reply by toammann** [reply]
> existing Conducting Sheet model (which is based on the paper [1]), which already models surface impedance

As far as I have understood, the surface impedance concept should apply also to 3D shaped conductors. This would be an improvement over the conducting sheet model. However, the fields have to treated differently at the material boundaries. 

I need definitely more theory on this. A method which supports 3D shapes should be preferred (if the implementation effort is reasonable).

**Comment by S1J2** [comment]
Hello All,
I went thru your discussion on topic and shared examples of model fitting, which looks well promising. Im interested into use case of signal integrity application - wideband S-parameters including both conductor and dielectric loss or maybe just characterization of multilayer via or  connector with breakout region. For such tasks it seems to me that OpenEMS is not completely suitable do to TD nature but as workaround I was thinking if it would be possible and reasonable to resolve by "brute force". Such procedure could be based on running frequency sweep with harmonic excitation and material properties would change for each frequency point according pre-calculated model fit, obviously it would result in lot of simulation runs and maybe some algorithm to find proper/minimum number of sweep points can be deployed, Im not sure if its possible to reuse some parts of simulation result to minimize calculation in each sweep to just necessary minimum, but it should yield precise S-parameters values for each frequency point which was simulated, including freq. dependent loss. Or is it completely wrong direction how to handle this topic?

**Reply by biergaizi** [reply]
> Such procedure could be based on running frequency sweep with harmonic excitation and material properties would change for each frequency point according pre-calculated model fit, obviously it would result in lot of simulation runs [...] but it should yield precise S-parameters values for each frequency point which was simulated, including freq. dependent loss. Or is it completely wrong direction how to handle this topic?

I think it can work, at least for narrowband applications. If you only need a few data points, why not? I've previously used this idea to simulate the effect of a thin layer of solder mask (which is difficult to model in FDTD) - first I run a 2D simulation to find its influence to the characteristic impedance, then I "fudge" my openEMS model with an "unphysical" layer on thick dielectric, I then adjust its dielectric constant until I get the same characteristic impedance as the thin solder mask. If it's just an one-off task, it's good enough.

But in the general case, this solution is abusive. The strength of FDTD is creating a wideband simulation in time domain (remember the native output of FDTD and openEMS is a time-varying signal waveform, S-parameters are obtained by Fourier transform). All frequency-dependent effects should be included in the material modeling code, otherwise for wideband simulations the time-domain results will be unusable. The mixing of S-parameter data obtained from different simulation runs would probably also violate [passivity and causality](https://dl.cdn-anritsu.com/en-us/test-measurement/ohs/10450-00040AD/index.html#page/Calibration_and_Measurement/MG_Ch09_AdapterRemoval.15.7.html), making it unusable for circuit simulators for wideband uses, and signal integrity simulations like eye diagrams are wideband simulations. If you're using FDTD for modeling one frequency at a time, it would be the wrong tool for the job. So I believe it's crucial that the community develops better material and conductor models as I proposed in the original post.
